### DISCUSSION TITLE
我的对象存储密钥会被上传吗

### DISCUSSION THREAD

**Discussion by simonla** [Kuingsmile/PicHoro#27] [Q&A] [answered]
看到app里有登录功能，以及云端拉取配置功能

担心我的密钥被上传到第三方的服务器

**Comment by Kuingsmile** [comment, marked as answer]
> 看到app里有登录功能，以及云端拉取配置功能
> 
> 担心我的密钥被上传到第三方的服务器

@simonla 
为了实现云端功能，密钥保存于我的个人服务器内搭建的MySQL数据库上，但在保存时，使用了加密存储，加密时进行了加盐操作，加密的密钥根据您的登录密码md5/sha1值生成，登录密码保存于本地，除了您的用户名在后台是明文保存，其它所有信息都是加密存储的，我看不到任何信息。如果您不主动或被动泄露了密码，即使拿到我的数据库文件所有内容，也不会被破解得到您的对象存储密钥等信息。

此外，实际使用的加密的方法我并没有上传到Github上，您可以参考`lib/utils/sql_utils.dart`文件来查看我保存sql的代码，但里面的encryptSelf和decryptSelf两个函数只是早期demo版本的示例，实际并没有在release版本中使用。

以下是我自己的账号保存的信息的截图：
![mysql_lankong_test](https://user-images.githubusercontent.com/96409857/201035958-f8420e58-2d02-48b6-b58d-953744aa39a7.png)

该云端功能只是为了方便用户在多台手机或更换手机、卸载重新安装软件后快速使用。如果您对这个功能有介意，可以连接您的个人数据库地址，然后重新打包软件，安装使用哈~ :smile:
